docs/en/03_Upgrading/02_Upgrading_module.md

```diff
@@ -0,0 +1,331 @@
+Making your module compatible with SilverStripe 4 is only one part of the process. As a module maintainer, you also want to provide a good upgrade experience for your users. Your module can integrate with the [SilverStripe upgrader](https://github.com/silverstripe/silverstripe-upgrader) just like the SilverStripe core modules.
+
+Your SilverStripe 4 module should ship with a `.upgrade.yml` file. This file is read by the upgrader and will define new APIs introduced by the upgraded version of your module. Each steps in this guide details what entry you should add to your module's `.upgrade.yml` file.
+
+## Step 0 - Branching off your project
+
+You'll want to run your module upgrade on a dedicated development branch. While it's possible to upgrade a module from within a SilverStripe project, it's usually cleaner and easier to clone your module and work directly on it.
+
+```bash
+# We're assumming that the default branch of you module is the latest SS3 compatible branch 
+git clone git@github.com:example-user/silverstripe-example-module.git
+cd silverstripe-example-module
+
+git checkout -b pulls/ss4-upgrade
+git push origin pulls/ss4-upgrade --set-upstream
+```
+
+If you're planning to keep supporting the SilverStripe 3 version of your module, consider creating a dedicated SilverStripe 3 branch.
+
+To require the development branch of your module in a SilverStripe 4 project, you can use composer and prefix the name the name of your branch with `dev-`.
+
+```bash
+composer require example-user/silverstripe-example-module dev-pulls/ss4-upgrade
+```
+
+If the development branch is hosted on a different GIT remote than the one used to publish your module, you'll need to add a VCS entry to your test project's `composer.json` file.
+
+```json
+{
+  "name": "example-user/test-project",
+  "type": "project",
+  "require": {
+    "example-user/silverstripe-example-module": "dev-pulls/ss4-upgrade"
+  },
+  "repositories": [
+    {
+      "type": "vcs",
+      "url": "git@github.com:alternative-user/silverstripe-example-module.git"
+    }
+  ]
+}
+```
+
+You will not be able to install your development branch in a SilverStripe 4 project until you've adjusted your module's dependencies.
+
+## Step 1 - Upgrade your dependencies
+
+Before you can install your module in a SilverStripe 4 project, you must update your module's `composer.json` file to require SilverStripe 4 compatible dependencies. In most cases, you'll be better off updating your module's composer file manually, especially if your module only requires a small number of dependencies. You can use upgrader's `recompose` command if you want, but you'll need to carefully validate the resulting `composer.json` file.
+
+### Update module's type 
+
+SilverStripe 4 modules are now installed inside the vendor directory. To get your module installed in the vendor directory, you'll need to update its `type` to `silverstripe-vendormodule`. You'll also need to add a dependency to `silverstripe/vendor-plugin`.
```

Comment:
I don't believe this is true. As long as you have a framework dependency you don't need the vendor module plugin.

Reply:
Strictly speaking, you're right as all modules will directly or indirectly require framework. However, all of our own modules explicitly require `silverstripe/vendor-plugin`, so this kind of sounds like a best practice to me.